src/components/atoms/ToggleSwitch/ToggleSwitch.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { ChangeEvent, MouseEvent } from 'react';
+
+import './toggleSwitch.css'
+
+const ToggleSwitch = (props: Props) => {
+  const {
+    switchCircleColor, activeBackgroundColor, width,
+    inactiveBackgroundColor, onChange, height,
+    isActive, leftText, rightText
+  } = props;
+
+  // circleDiameter is the diameter of the circular slider which should be smaller than the size of the parent component so as to provide offset between the slider and it's parent
+  const circleDiameter = height - 4;
+
+  const switchLabelStyle = {
+    background: isActive ? activeBackgroundColor : inactiveBackgroundColor,
+    width,
+    height,
+    borderRadius: height,
+    top: '0px', // required here, as somewhere in the project(globally) these values are altered
+    left: '0px' // required here
+  }
+  const switchDivStyle = {
+    width,
+    height
+  }
+  const inputStyle = {
+    margin: '0px' // required here, as somewhere in the project(globally) these values are altered
+  }
+
+  const switchButtonStyle = {
+    background: switchCircleColor,
+    width: `${circleDiameter}px`,
+    height: `${circleDiameter}px`,
+    transform: isActive ? `translateX(${width - circleDiameter - 4}px)` : 'none'
+  }
+
+  return (
+    <div className="valign-wrapper">
+      {leftText}
+      <div style={switchDivStyle} className='sw348reactSwitchDivision' onClick={(e) => onChange(e)}>
```

Comment:
Don't we get 2 onChange events in parent here? one from this onClick and one from input tag's onChange?